lstchain/io/io.py

```diff
@@ -268,22 +267,29 @@ def merging_check(file_list):
     assert len(file_list) > 1, "The list of files is too short"
     mergeable_list = file_list.copy()
 
-    filename0 = mergeable_list[0]
-    array_info0 = read_array_info(filename0)
-    mcheader0 = read_simu_info_hdf5(filename0)
-    thrown_events_hist0 = read_simtel_energy_histogram(filename0)
-    metadata0 = read_metadata(filename0)
+    first_file = mergeable_list[0]
+    subarray_info0 = SubarrayDescription.from_hdf(first_file)
+    metadata0 = read_metadata(first_file)
+
+    if subarray_info0.name == "MonteCarloArray":
+        mcheader0 = read_simu_info_hdf5(first_file)
+        thrown_events_hist0 = read_simtel_energy_histogram(first_file)
+
     for filename in mergeable_list[1:]:
         try:
-            mcheader = read_simu_info_hdf5(filename)
-            thrown_events_hist = read_simtel_energy_histogram(filename)
             metadata = read_metadata(filename)
             check_metadata(metadata0, metadata)
-            check_mcheader(mcheader0, mcheader)
-            check_thrown_events_histogram(thrown_events_hist0, thrown_events_hist)
-            for ii, table in read_array_info(filename).items():
-                assert (table == array_info0[ii]).all()
-        except:
+            subarray_info = SubarrayDescription.from_hdf(filename)
+
+            if subarray_info0.name == "MonteCarloArray":
+                mcheader = read_simu_info_hdf5(filename)
+                thrown_events_hist = read_simtel_energy_histogram(filename)
+                check_mcheader(mcheader0, mcheader)
+                check_thrown_events_histogram(thrown_events_hist0, thrown_events_hist)
+
+            assert subarray_info == subarray_info0
```

Comment:
Also, I'm not quite sure of this assertion. I guess it checks that all the tables are the same. Before it was done by a loop.